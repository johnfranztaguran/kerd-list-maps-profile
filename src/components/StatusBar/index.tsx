import React from 'react'
import { StatusBar as RNStatusBar } from 'react-native'
import { themes } from 'constant'

const StatusBar = (props: {
  hidden?: boolean
  translucent?: boolean
  networkActivityIndicatorVisible?: boolean
  backgroundColor?: string
  statusBarStyle?: 'dark-content' | 'default' | 'light-content'
}) => {
  const {
    hidden = false,
    translucent = false,
    networkActivityIndicatorVisible = true,
    statusBarStyle = 'light-content',
    backgroundColor = themes.colors.primary
  } = props
  return (
    <RNStatusBar
      // dark-content, light-content and default
      barStyle={statusBarStyle}
      // To hide statusBar
      hidden={hidden}
      // allowing light, but not detailed shapes
      translucent={translucent}
      // The background color of the status bar. Android-only
      backgroundColor={backgroundColor}
      // Control the visibility of the network activity indicator. iOS-only.
      networkActivityIndicatorVisible={networkActivityIndicatorVisible}
    />
  )
}

export default StatusBar
