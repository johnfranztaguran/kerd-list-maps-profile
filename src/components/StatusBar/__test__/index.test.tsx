import React from 'react'
import { render } from '@testing-library/react-native'
import StatusBar from '../index'

describe('Component: StatusBar', () => {
  it('Should render a default <StatusBar>', () => {
    const { baseElement } = render(<StatusBar />)
    expect(baseElement).toMatchSnapshot()
  })

  it('Should render a with props <StatusBar>', () => {
    const { baseElement } = render(
      <StatusBar
        translucent
        backgroundColor="red"
        statusBarStyle="light-content"
      />
    )
    expect(baseElement).toMatchSnapshot()
  })
})
