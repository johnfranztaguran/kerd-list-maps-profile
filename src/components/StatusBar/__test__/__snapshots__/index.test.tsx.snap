// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: StatusBar Should render a default <StatusBar> 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`Component: StatusBar Should render a with props <StatusBar> 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
